pkg/detector/detector.go

```diff
@@ -794,7 +794,17 @@ func (d *ResourceDetector) OnPropagationPolicyAdd(obj interface{}) {
 
 // OnPropagationPolicyUpdate handles object update event and push the object to queue.
 func (d *ResourceDetector) OnPropagationPolicyUpdate(oldObj, newObj interface{}) {
-	// currently do nothing, since a policy's resource selector can not be updated.
+	if resourceSelectorsDeepEqual(oldObj, newObj) {
+		return
+	}
```

Comment:
Since the `detector` only focuses on the `resource` match thing instead of `cluster` match. 
That gives an illusion that here we just care about the `ResourceSelector` would be enough, but we still have other fields that need to be updated when building ResourceBinding, like ([PropagateDeps](https://github.com/karmada-io/karmada/blob/07ddc07997224c1ea4df21ce65b0ac8b2f0e22b5/pkg/detector/detector.go#L723)), so my opinion is to remove this check, just put all PP into the queue.

Reply:
Update.